### PR TITLE
Simulator codebase integration

### PR DESCRIPTION
**Pull request by devartstar**
Fixes : #3 

In this PR,

1.  Imported the simulator code from the CircuitVerse repository.

2.  I used my GSoC task's dummy Vue Component from simulator HTML. 
So Navbar is already broken down into smaller components 
(Options API needs to be updated to Composition API and some other changes to be done).

3.  Imported simulator stylesheets into styles folder. 
     Styles will be refactored later to its specefic components.

4.  jQuery UI not working so I commented those elements. 
So accordion, dialogBox, draggable, and resizable not working.
Will replace these using Vuetify.

5.  Added pre-commit-hook `Husky` and `Commitlint` to check on the commits.

6.  Redirected `/` route to `/simulator`

7.  Added dependencies necessary for the simulator to be working.
(added Bootstrap temporarily)

The simulator's core functionalities are working properly.

https://user-images.githubusercontent.com/61665451/174111015-443cc62e-e8b7-454e-a23a-e5a74dba2d34.mp4

**To-do :** Fix EsLint errors in simulator javascript files.

### CODE REVIEW

**Comment by tachyons** [comment]
Could you try if it is possible to move code from main repository preserving the commits, One approach is [this](https://medium.com/@ayushya/move-directory-from-one-repository-to-another-preserving-git-history-d210fa049d4b)

**Comment by devartstar** [comment]
@tachyons @abstrekt @ayan-biswas0412 can you please review this PR.

**Comment by tachyons** [comment]
Could you check why images aren't loading on preview? rest of issues can be fixed later ?

**Comment by tachyons** [comment]
https://deploy-preview-4--circuitverse.netlify.app

**Comment by devartstar** [comment]
> Could you check why images aren't loading on preview? rest of issues can be fixed later ?

Fixed it, error was because I was using absolute paths for images as relative path wasn't working before.
But used [this doc](https://vitejs.dev/guide/assets.html#new-url-url-import-meta-url) to make relative paths work.

**Review by tachyons — APPROVED** [review]

**Review by ayan-biswas0412 — APPROVED** [review]
Good work @devartstar

File: src/components/Navbar/User/User.vue [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                href="https://circuitverse.org/users/sign_in"
+            >
+                Sign In
+            </a>
+        </li>
+    </ul>
+</template>
+
+<script>
+import DropDown from '../../Dropdown/DropDown.vue'
```
Comment (abstrekt) [inline src/components/Navbar/User/User.vue:42]: why are we not using aliases?

File: src/simulator/spec/testData/gates-testdata.json [hunk 2]
```diff
@@ -0,0 +1,200 @@
+{
```
Comment (abstrekt) [inline src/simulator/spec/testData/gates-testdata.json:1]: we'll need to check what all constants/configs we have and move them to /constants and import from there.

Comment (abstrekt) [reply src/simulator/spec/testData/gates-testdata.json:1]: but we can discuss this later.

**Review by abstrekt — APPROVED** [review]

File: src/simulator/spec/testData/gates-testdata.json (hunk 2 above)
Comment (devartstar) [reply src/simulator/spec/testData/gates-testdata.json:1]: Okay

**Pull request merged**